db/models/aggregate_functions.py

```diff
@@ -153,9 +153,31 @@ def submit_application(application_id):
 def update_form(application_id, form_name, question_json):
     try:
         form_sql_row = FormsMethods.get_form(application_id, form_name)
+        # Running update form for the first time
+        if question_json and not form_sql_row.json:
+            ApplicationsMethods.application_edited(application_id)
+            current_app.logger.info(
+                f"Application updated for application_id: '{application_id}."
+            )
```

Comment:
each code block within these if else statements appears to be doing the same thing?

Reply:
yes so all the block does is update the 'last_edited_on' field and we only want that to happen in these 3 scenarios (the comments above each if/ elif should say what that scenario would be) :)

Reply:
okay so if we have a legitimate case for update we run the update 👍 so does this also imply that we should only run the finally block in these cases also
```
        form_sql_row.json = question_json
        update_statuses(application_id, form_name)
        db.session.commit()
```
in other words all this content could go into ApplicationsMethods.application_edited() (maybe rename) and in the case its not a legit update we log/throw? What do you think?

Reply:
so the finally block will run regardless of what happens i.e. even if these conditions aren't met the finally block will still be run! So think your point on moving some of the finally block logic into the application edited method is a great shout as it should only really be ran there 👌🏾 but the update statuses function lives in the aggregate functions file so it will introduce a circular import error 😟 I can move the form_sql_row = question_json line though if we think it's worth it? otherwise can leave as is and it won't change functionality?

Reply:
I have moved the method from application.py to aggregate functions.py to avoid the circular import and have been able to incorporate this^ 😄